pkg/sip/inbound.go

```diff
@@ -696,6 +753,71 @@ func (c *inboundCall) handleDTMF(tone dtmf.Event) {
 	}
 }
 
+func (c *inboundCall) transferCall(ctx context.Context, transferTo string) error {
+	err := c.cc.transferCall(ctx, transferTo)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case <-time.After(30 * time.Second):
+		return psrpc.NewErrorf(psrpc.DeadlineExceeded, "refer timed out")
+	case <-ctx.Done():
+		return psrpc.NewErrorf(psrpc.Canceled, "refer canceled")
+	case err := <-c.referDone:
+		if err != nil {
+			return err
+		}
+	}
+
+	c.cc.sendBye()
+
+	// This is needed to actually terminate the session before a media timeout
+	c.Close()
+
+	return nil
+
+}
+
+func (c *inboundCall) handleNotify(req *sip.Request, tx sip.ServerTransaction) error {
+	method, cseq, status, err := handleNotify(req)
+	if err != nil {
+		return err
+	}
+
+	switch method {
+	case sip.REFER:
+		c.cc.mu.RLock()
+		defer c.cc.mu.RUnlock()
+
+		if cseq != 0 && cseq != uint32(c.cc.referCseq) {
+			// NOTIFY for a different REFER, skip
+			return nil
+		}
```

Comment:
Sure. For reference, what is the slip between sipInbound and inboundCall? I put this into inboundCall because this is part of the call workflow, which seemed usually handled by inboundCall (vs signaling and call parameters in sipInbound)

Reply:
My thinking was that all SIP protocol details should be hidden in `sipInbound`. That way we could test it separately and eventually update sipgo package to the latest, all without changing the `inboundCall` logic.